### PR TITLE
v2.0 Release

### PR DESCRIPTION
**Pull request by glenn-jocher**
Signed-off-by: Glenn Jocher <glenn.jocher@ultralytics.com>

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
YOLOv5 major updates with enhanced model definitions and performance improvements.

### 📊 Key Changes
- Updated YOLOv5 pretrained checkpoints with slightly different performance metrics (AP values and model names).
- Modified `Detect` layer initialization to accommodate new channel count parameter (`ch`).
- Refined anchor order check to ensure consistency with stride order.
- Adjusted values in training hyperparameters, such as class loss gain and HSV augmentation strengths.
- Implemented a cosmetic change to `README.md` on model public release and development start dates.
- Modified LR scheduler function for a new learning rate cosine annealing strategy.
- Increased balancing factors in loss computation to tweak the influence of each prediction head.
- Changed batch normalization momentum and epsilon values in `utils/torch_utils.py`.

### 🎯 Purpose & Impact
- 🚀 **Enhanced performance:** Tweaks in model and loss configuration can lead to more accurate detection and training stability.
- 🤖 **Model training:** Hyperparameter adjustments and learning rate scheduler optimization for better convergence.
- 👀 **Clearer documentation:** README update helps users to follow the timeline of significant changes and releases.
- 🧠 **Backend improvements:** Code modifications and cleanup contribute to maintenance and potential performance gains.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
This v2.0 release PR is complete, all CI CPU tests are passing. Will merge and subsequently issue v2.0 release. 

IMPORTANT: v2.0 release contains breaking changes. Models trained with earlier versions will not operate correctly with v2.0. The last commit before v2.0 that operates correctly with all earlier pretrained models is:
https://github.com/ultralytics/yolov5/tree/5e970d45c44fff11d1eb29bfc21bed9553abf986

**Comment by glenn-jocher** [comment]
Commands to clone last commit prior to v2.0:

```bash
git clone https://github.com/ultralytics/yolov5  # clone repo
cd yolov5
git reset --hard 5e970d4  # last commit before v2.0
```
